### PR TITLE
Toolbox

### PR DESCRIPTION
**Pull request by fnevgeny**
This implements the #302 feature request - a floating toolbox with GUI controls for a subset of actions (currently, togglePointer, toggleDrawing, toggleEraser, freeze, blank, hide, pause, as well as pen/eraser size and pen color). The direction and initial state of the toolbox can be set by the toolbox-direction and toolbox-state pdfpcrc config file options.

### CODE REVIEW

File: man/pdfpcrc.in [hunk 1]
```diff
@@ -9,9 +9,6 @@
 .SH NAME
 pdfpcrc \- pdfpc configuration file
 
-.SH SYNOPSIS
```
Comment (AndreasBilke) [inline man/pdfpcrc.in]: This should rather go to a FILES sections?

File: man/pdfpcrc.in [hunk 2]
```diff
@@ -92,6 +89,13 @@ Enable color hints of the timer, continuously indicating whether the
 presentation is progressing according to the expected pace (bool, Default true).
 See the timer operation description in
 .B pdfpc(1).
+.TP
+.B toolbox-direction
+Set the toolbox direction (LtoR, RtoL, TtoB, or BtoT for the left-to-right,
+right-to-left, top-to-bottom, or botton-to-top, respectively). Default is LtoR.
+.TP
+.B toolbox-state
+Set the toolbox state (Shown|Hidden|Minimized). Default is Minimized.
```
Comment (AndreasBilke) [inline man/pdfpcrc.in]: I would prefer to auto-hide it, or do some tablet detection.

File: src/classes/window/presenter.vala [hunk 3]
```diff
@@ -153,7 +184,83 @@ namespace pdfpc.Window {
          **/
         protected int next_allocated_width;
 
-        /**
+        protected bool on_button_press(Gtk.Widget pbut, Gdk.EventButton event) {
+	    if (event.button == 1 ) {
+                var w = this.get_window();
+
```
Comment (AndreasBilke) [inline src/classes/window/presenter.vala]: This looks unaligned.

Comment (fnevgeny) [reply src/classes/window/presenter.vala]: Seems to me OK. There are tab characters (probably copy&paste'd from another function), but there are many other places in the pdfpc sources. Maybe we need to run everything through expand once?

File: man/pdfpcrc.in (hunk 1 above)
Comment (fnevgeny) [reply man/pdfpcrc.in]: You are right, yes.

File: man/pdfpcrc.in (hunk 2 above)
Comment (fnevgeny) [reply man/pdfpcrc.in]: In fact, I think it is also usable for non-tablet hardware. Especially for new users, who have not yet memorized keyboard shortcuts. (Well, I personally use it :).) As such, if it is hidden by default, the new feature will remain unnoticed. On the other hand, if shown by default and somebody finds it annoying, it is a matter of adding a single config statement to pdfpcrc...

File: man/pdfpcrc.in (hunk 1 above)
Comment (fnevgeny) [reply man/pdfpcrc.in]: Please check now.

**Comment by fnevgeny** [comment]
About the icons/settings.svg license: [https://openclipart.org/share](https://openclipart.org/share).

**Comment by fnevgeny** [comment]
Re: the slider issue should be fixed now.

File: man/pdfpcrc.in (hunk 2 above)
Comment (fnevgeny) [reply man/pdfpcrc.in]: OK, let's make the toolbox hidden by default. I then suggest to add toggleToolbox action and bind it, say, to "t".

**Pull request merged**